### PR TITLE
[4.0] Fix Legacy Model loading. Fixes #23517

### LINKED ISSUE #23517: [4.0] Legacy View not working - no data from model
### Steps to reproduce the issue
When you open some view in administration which is based on JViewLegacy (external component views), it does not load data from model

### Expected result
Data from model are loaded.

### Actual result
Joomla! 4.0.0-alpha6 - everything works as expected.
Joomla! 4.0.0-alpha7-dev - data are not loaded.

You can test it with some external component view or build your own:
```php
jimport( 'joomla.application.component.view' );

class SomeView extends JViewLegacy
{
	protected $state;

	function display($tpl = null) {

		$this->state		= $this->get('State');
		dump($this->state);
	}
}
```

produces in Alpha6
OK

but in Alpha7 (latest build created by composer/npm):
OK

Data from model are not loaded at all
### System information (as much as possible)
Joomla! 4.0.0-alpha6 - works OK :heavy_check_mark: 
Joomla! 4.0.0-alpha7-dev (latest build created by composer/npm) - data are not loaded. :red_circle: 

Joomla! Version | 4.0.0-alpha7-dev
-- | --
PHP Version | 7.2.4
Identity | user
Response | 200
Template | atum
Database | ServermysqlVersion5.5.5-10.2.6-MariaDBCollationutf8mb4_unicode_ciConn Collationutf8_general_ci

### PR DESCRIPTION
**Pull request by wilsonge**
See #23517 for test instructions. Note you must test with an extension that hasn't been namespaced and uses core MVC

### CODE REVIEW

File: libraries/src/MVC/Controller/BaseController.php [hunk 1]
```diff
@@ -181,7 +182,7 @@ class BaseController implements ControllerInterface
 	 */
 	public static function addModelPath($path, $prefix = '')
 	{
-		BaseDatabaseModel::addIncludePath($path, $prefix);
+        BaseModel::addIncludePath($path, $prefix);
```
Comment (Quy) [inline libraries/src/MVC/Controller/BaseController.php]:
```suggestion
		BaseModel::addIncludePath($path, $prefix);
```

**Review by laoneo — APPROVED** [review]

**Comment by PhocaCz** [comment]
I have tested this item :white_check_mark: successfully on efbeebbcac30555c90146529ab28d27e78124741<br /><br />Tested on Joomla! 4.0.0-alpha7-dev<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23525">issues.joomla.org/tracker/joomla-cms/23525</a>.</sub>

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on efbeebbcac30555c90146529ab28d27e78124741<br /><br />Tested on alpha-7 with new installation of extension Phocadownload.
But: I did'nt get the failure as shown in #23525. Phocadownload > Files, I got:
An error has occurred.

    0 Call to a member function get() on null 
After using the Patch, above message has gone.
 <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23525">issues.joomla.org/tracker/joomla-cms/23525</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23525">issues.joomla.org/tracker/joomla-cms/23525</a>.</sub>

**Comment by ChristineWk** [comment]
Checked again, because first test was only with "Files" but now, all: (present status):
Files / Categories /  Licenses / Downloads / Uploads / File Rating / Styles
0 Call to a member function get() on null 

Layout
Call to a member function getTable() on boolean and:
Warning: Invalid argument supplied for foreach() in /.../administrator/components/com_phocadownload/views/phocadownloadstat/view.html.php on line 28 Warning: count(): Parameter must be an array or an object that implements Countable in .../administrator/components/com_phocadownload/views/phocadownloadstat/view.html.php on line 36 

after Patch; both messages has gone

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23525">issues.joomla.org/tracker/joomla-cms/23525</a>.</sub>

**Comment by wilsonge** [comment]
Thanks guys :)

**Pull request merged**